### PR TITLE
Add Interprter.version_info and version test cases

### PR DESCRIPTION
**Pull request by brettlangdon**
Add new `Interpreter.version_info` function and some version test cases.

### CODE REVIEW

File: riot/riot.py [hunk 1]
```diff
@@ -86,6 +86,15 @@ def version(self) -> str:
         version = output.decode().strip().split(" ")[1]
         return version
 
+    @functools.lru_cache()
+    def version_info(self) -> t.Tuple[int, int, int]:
+        version_head = "".join(
+            list(itertools.takewhile(lambda _: _.isdigit() or _ == ".", self.version()))
```
Comment (jd) [inline riot/riot.py:92]: Can we use something like `python -c 'import sys; print("%s.%s.%s" % (sys.version_info.major, sys.version_info.minor, sys.version_info.micro))'` in  the`version()` and something like that here?

Sounds safer than parsing strings.

Comment (brettlangdon) [reply riot/riot.py:92]: sounds good! on it!

File: tests/test_unit.py [hunk 2]
```diff
@@ -26,3 +34,16 @@ def test_interpreter(v1, v2, equal):
         assert Interpreter(v1) != Interpreter(v2)
         assert hash(Interpreter(v1)) != hash(Interpreter(v2))
         assert repr(Interpreter(v1)) != repr(Interpreter(v2))
+
+
+def test_interpreter_version(current_interpreter: Interpreter) -> None:
+    version, _, _ = sys.version.partition(" ")
```
Comment (brettlangdon) [inline tests/test_unit.py:40]:
```suggestion
    version = "%s.%s.%s" % (sys.version_info.major, sys.version_info.minor, sys.version_info.micro)
```

File: riot/riot.py [hunk 3]
```diff
@@ -81,10 +81,19 @@ def __str__(self) -> str:
...
+                "-c",
+                'import sys; print("%s.%s.%s" % (sys.version_info.major, sys.version_info.minor, sys.version_info.micro))',
+            ],
+        )
+        return output.decode().strip()
+
+    @functools.lru_cache()
+    def version_info(self) -> t.Tuple[int, int, int]:
+        version_info = tuple(int(_) for _ in self.version().split(".")) + (0, 0, 0)
+        return t.cast(t.Tuple[int, int, int], version_info[:3])
```
Comment (jd) [inline riot/riot.py:96]:
```suggestion
        return tuple(map(int, self.version().split("."))
```
no?

Comment (brettlangdon) [reply riot/riot.py:96]: will still need the cast, but yeah, this is cleaner. thanks!

File: tests/test_unit.py [hunk 4]
```diff
@@ -26,3 +34,20 @@ def test_interpreter(v1, v2, equal):
...
+    )
+    assert current_interpreter.version() == version
+
+
+def test_interpreter_version_info(current_interpreter: Interpreter) -> None:
+    assert current_interpreter.version_info() == (
+        sys.version_info[0],
+        sys.version_info[1],
+        sys.version_info[2],
+    )
```
Comment (jd) [inline tests/test_unit.py:53]:
```suggestion
    assert current_interpreter.version_info() == sys.version_info[:3]
```

**Review by jd — APPROVED** [review]

**Pull request merged**